lib/msal-common/src/account/AuthToken.ts

```diff
@@ -47,4 +47,27 @@ export class AuthToken {
             throw ClientAuthError.createTokenParsingError(err);
         }
     }
+
+    /**
+     * Determine if the token's max_age has transpired
+     */
+    static checkMaxAge(authTime: number | undefined, maxAge: number): void {
+        /*
+         * per https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest
+         * To force an immediate re-authentication: If an app requires that a user re-authenticate prior to access,
+         * provide a value of 0 for the max_age parameter and the AS will force a fresh login.
+         */
+        if (maxAge === 0) {
+            throw ClientAuthError.createMaxAgeTranspiredError();
+        }
+
+        if (!authTime) {
```

Comment:
Feels weird to allow this to be undefined in the function definition but then throw if it's undefined. I'd suggest making this parameter required here and throwing this error at the callsite if it's not defined.